### PR TITLE
Timetable 

### PR DESCRIPTION
**Pull request by ravi-kishan**
First steps for #144 

Changes: Implemented timetable feature in app.
TODO: UI modification

### CODE REVIEW

**Review by dr0pdb — CHANGES_REQUESTED** [review]
@ravi-kishan Please fix the codacy issues.

File: app/src/main/java/com/iitp/njack/iitp_connect/core/profile/ProfileActivity.java [hunk 1]
```diff
@@ -25,16 +25,17 @@
...
     private ActivityProfileBinding binding;
     private ProfileViewModel profileViewModel;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
+        ArrayAdapter<CharSequence> courseAdapter;
+        ArrayAdapter<CharSequence> branchAdapter;
+        ArrayAdapter<CharSequence> yearAdapter;
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/profile/ProfileActivity.java]: Don't do this. The previous one was correct. This is not desirable.

**Review by dr0pdb — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/iitp/njack/iitp_connect/core/profile/ProfileRepository.java [hunk 2]
```diff
@@ -12,10 +12,9 @@
 import javax.inject.Inject;
 
 public class ProfileRepository {
-    private DatabaseReference databaseReference;
-
     @VisibleForTesting
-    FirebaseDatabaseLiveData firebaseDatabaseLiveData;
+    private FirebaseDatabaseLiveData firebaseDatabaseLiveData;
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/profile/ProfileRepository.java]: Make it public. It is causing the build failure.

**Review by dr0pdb — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/iitp/njack/iitp_connect/core/timetable/TimeTableActivity.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+public class TimeTableActivity extends AppCompatActivity implements View.OnClickListener {
+    @Inject
+    public ViewModelProvider.Factory viewModelFactory;
+    @Inject
+    public User user;
+    @Inject
+    public TimeTableInformation timeTableInformation;
+
+    TimeTableViewModel timeTableViewModel;
```
[the remark this answers is not in the document]
Comment (srv-twry) [reply app/src/main/java/com/iitp/njack/iitp_connect/core/timetable/TimeTableActivity.java]: @ravi-kishan Make it private.

[the remark this answers is not in the document]
Comment (ravi-kishan) [reply app/src/main/java/com/iitp/njack/iitp_connect/core/timetable/TimeTableActivity.java]: Dagger inject does not work on private fields.

[the remark this answers is not in the document]
Comment (srv-twry) [reply app/src/main/java/com/iitp/njack/iitp_connect/core/timetable/TimeTableActivity.java]: Ok, then make it public. Explicitly mention it.

**Comment by dr0pdb** [comment]
check the build and fix it.

**Review by dr0pdb — APPROVED** [review]

**Pull request merged**